crates/doc/src/builder.rs

```diff
@@ -349,10 +349,23 @@ impl DocBuilder {
             .unwrap()
             .insert(String::from("title"), self.config.title.clone().into());
         if let Some(ref repo) = self.config.repository {
+            // Create the full repository URL
+            let git_repo_url = match &self.config.path {
+                Some(path) => {
+                    // If path is specified, append it to the repository URL
+                    let repo_clean = repo.trim_end_matches('/');
+                    format!("{}/{}", repo_clean, dir.trim_start_matches('/'))
+                }
+                None => {
+                    // No path specified, use repository URL as-is
+                    repo.clone()
+                }
+            };
```

Comment:
this makes sense,

I'd prefer a `if let some else` style here instead of the match